providers/base.ts

```diff
@@ -351,6 +351,48 @@ export abstract class ReleaseLookup<Provider extends MetadataProvider, RawReleas
 	}
 }
 
+/** Extends MetadataProvider with functions common to lookups accessing web APIs. */
+export abstract class MetadataApiProvider extends MetadataProvider {
+	/** Must be implemented to perform a request against the specified URL. */
+	abstract query<Data>(apiUrl: URL, maxTimestamp?: number): Promise<CacheEntry<Data>>;
+}
+
+/** Extends ReleaseLookup with functions common to lookups accessing web APIs. */
+export abstract class ReleaseApiLookup<Provider extends MetadataApiProvider, RawRelease>
+	extends ReleaseLookup<Provider, RawRelease> {
+	/** Constructs an API URL for a release using the specified lookup options. */
+	abstract constructReleaseApiUrl(): URL;
+
+	/** Performs the query for the URL returned by constructReleaseApiUrl for all configured regions. */
+	protected async queryAllRegions<Data>(
+		dataValidator: (data: Data) => boolean,
+		errorValidator: (error: unknown) => boolean = (_) => false,
```

Comment:
Without reading the implementation, I have an intuition of the meaning of `true` and `false` for the data validator, but not for the error validator.
Making the names more verbose might help:
```suggestion
		dataValidator: (data: Data) => boolean,
		ignoreError: (error: unknown) => boolean = (_) => false,
```
Or with inverse meaning:
```suggestion
		isValidData: (data: Data) => boolean,
		isCriticalError: (error: unknown) => boolean = (_) => true,
```